exense-commons-parent/exense-basic-commons/src/main/java/ch/exense/commons/processes/ExternalJVMLauncher.java

```diff
@@ -16,59 +16,97 @@
 		String[] classPathEntries = System.getProperty("java.class.path").split(File.pathSeparator);
 
-		StringBuilder cp = new StringBuilder();
-		cp.append("\"");
-		for(String path:classPathEntries) {
-			cp.append(new File(path).getAbsolutePath()+File.pathSeparator);
+		File javaClassPathArgsFile = FileHelper.createTempFile();
+
+		try (OutputStreamWriter cp = new OutputStreamWriter(new FileOutputStream(javaClassPathArgsFile), StandardCharsets.UTF_8)) {
+			cp.append("-cp ");
+			cp.append("\"");
+			for(String path:classPathEntries) {
+				String absolutePath = new File(path).getCanonicalPath();
+				absolutePath = (isWindows()) ? absolutePath.replace("\\","/") : absolutePath;
+				cp.append(absolutePath);
+				cp.append(File.pathSeparator);
+			}
+			cp.append("\"");
 		}
-		cp.append("\"");
-		return cp.toString();
+		return javaClassPathArgsFile;
 	}
 
 	public ManagedProcess launchExternalJVM(String name, Class<?> mainClass, List<String> vmargs, List<String> progargs) throws ManagedProcessException {
 		return launchExternalJVM(name, mainClass, vmargs, progargs, true);
 	}
 	
 	public ManagedProcess launchExternalJVM(String name, Class<?> mainClass, List<String> vmargs, List<String> progargs, boolean redirectOutput) throws ManagedProcessException {
-		String cp = buildClasspath();
-		
-		List<String> cmd = new ArrayList<>();
-		cmd.add(javaPath);
-		cmd.add("-cp");
-		cmd.add(cp);
-		
-		cmd.addAll(vmargs);
-		
-		cmd.add(mainClass.getName());
-		
-		cmd.addAll(progargs);
-		
-		ManagedProcess process = new ManagedProcess(name, cmd, processLogFolder, redirectOutput);
-		process.start();
-		return process;
+		try {
+			File cpArgFile = buildClasspath();
+			List<String> cmd = new ArrayList<>();
+			cmd.add(javaPath);
+			cmd.add("@" + cpArgFile.getCanonicalPath());
+
+			cmd.addAll(vmargs);
+
+			cmd.add(mainClass.getName());
+
+			cmd.addAll(progargs);
+
+			ManagedProcess process = new JvmManagedProcess(cpArgFile, name, cmd, processLogFolder, redirectOutput);
+			process.start();
+			return process;
+		} catch (IOException e) {
+			throw new ManagedProcessException("Unable to create the java argument file specifying the classpath.",e);
+		}
 	}
 	public static boolean isWindows() {
-        return System.getProperty("os.name").toLowerCase().contains("win");
-    }
+		return System.getProperty("os.name").toLowerCase().contains("win");
+	}
+
+	private static class JvmManagedProcess extends ManagedProcess {
+
+		private File javaClassPathArgsFile;
+
+		public JvmManagedProcess(File javaClassPathArgsFile, String name, List<String> commands, File baseLogDirectory, boolean redirectOuput) throws ManagedProcessException {
+			super(name, commands, baseLogDirectory, redirectOuput);
+			this.javaClassPathArgsFile = javaClassPathArgsFile;
+		}
+
+		@Override
+		public void close() throws IOException {
+			super.close();
+			if (javaClassPathArgsFile != null && javaClassPathArgsFile.exists()) {
```

Comment:
If for some reason the process isn't closed by the client (i.e the method close() isn't called), the file would leak. For this reason I would add a deleteOnExit() in buildClasspath in addition to this